### PR TITLE
fix: Removed unused bucket field from connectors

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Removed unused `bucket` field from connectors
- Removed error handling code which has been made in the SDK

## Why

- This field was unused but accepted in the UI, it also confuses the user when we require a path for the files to process

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, this field was unused but accepted in the UI.

## Database Migrations

- No migrations performed for the existing records

## Notes on Testing

- Was able to add / edit a connector. Existing workflow ran fine too

## Screenshots
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/adapter_processor/adapter_processor.py [hunk 1]
```diff
@@ -94,12 +93,6 @@ def test_adapter(adapter_id: str, adapter_metadata: dict[str, Any]) -> bool:
             test_result: bool = adapter_instance.test_connection()
             logger.info(f"{adapter_id} test result: {test_result}")
             return test_result
-        # HACK: Remove after error is explicitly handled in VertexAI adapter
-        except json.JSONDecodeError:
-            raise TestAdapterInputError(
-                "Credentials is not a valid service account JSON, "
-                "please provide a valid JSON."
-            )
```
Comment (ritwik-g) [inline backend/adapter_processor/adapter_processor.py:102]: @chandrasekharan-zipstack why was this removed? And was this necessary for the change you were making?

File: unstract/connectors/src/unstract/connectors/filesystems/azure_cloud_storage/azure_cloud_storage.py [hunk 2]
```diff
@@ -65,9 +64,9 @@ def get_fsspec_fs(self) -> AzureBlobFileSystem:
     def test_credentials(self) -> bool:
         """To test credentials for Azure Cloud Storage."""
         try:
-            is_dir = bool(self.get_fsspec_fs().isdir(self.bucket))
+            is_dir = bool(self.get_fsspec_fs().isdir(""))
```
Comment (ritwik-g) [inline unstract/connectors/src/unstract/connectors/filesystems/azure_cloud_storage/azure_cloud_storage.py:67]: @chandrasekharan-zipstack what if the permissions provided are restrictive in nature and the access is only for particular bucket. Will this test connection still work? Should we may be include the path also in to this checks?

**Review by ritwik-g — COMMENTED** [review]
@chandrasekharan-zipstack I have left a couple of comments. Please check.

Also the original ticket mentions some UI changes. Is anyonre picking it up?

File: backend/adapter_processor/adapter_processor.py (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply backend/adapter_processor/adapter_processor.py:102]: Missed to add this in the PR description - this was from an earlier change which has been made in the [SDK](https://github.com/Zipstack/unstract-sdk/blob/4acfa684369f2d647b927d8d3bf92ee142a58ec0/src/unstract/sdk/adapters/llm/vertex_ai/src/vertex_ai.py#L89-L92)

**Comment by chandrasekharan-zipstack** [comment]
@ritwik-g regarding the UI changes needed - it'll take me sometime to read up on FE and implement. I don't think anyone else is assigned but I thought I'll make this BE only change atleast to help ease the UX

File: unstract/connectors/src/unstract/connectors/filesystems/azure_cloud_storage/azure_cloud_storage.py (hunk 2 above)
Comment (chandrasekharan-zipstack) [reply unstract/connectors/src/unstract/connectors/filesystems/azure_cloud_storage/azure_cloud_storage.py:67]: @ritwik-g we could take up testing access based on paths once we unify the BE / FE for the 2 columns (connector and source / dest configuration)

**Comment by ritwik-g** [comment]
> @ritwik-g regarding the UI changes needed - it'll take me sometime to read up on FE and implement. I don't think anyone else is assigned but I thought I'll make this BE only change atleast to help ease the UX

@chandrasekharan-zipstack okay. But please keep the ticket open in that case. @hari-kuriakose @nehabagdia we need to assign some one for the UI and UX changes.

File: backend/adapter_processor/adapter_processor.py (hunk 1 above)
Comment (ritwik-g) [reply backend/adapter_processor/adapter_processor.py:102]: Might have been a better idea to keep this as separate PR in that case.

**Review by ritwik-g — APPROVED** [review]

File: backend/adapter_processor/adapter_processor.py (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply backend/adapter_processor/adapter_processor.py:102]: Will take note

**Review by Deepak-Kesavan — APPROVED** [review]
